app/liquid_tags/stackery_tag.rb

```diff
@@ -1,44 +1,55 @@
 class StackeryTag < LiquidTagBase
   PARTIAL = "liquids/stackery".freeze
+  REGISTRY_REGEXP = %r{https://app.stackery.io/editor/design(?<params>\?.*)?}
+  PARAM_REGEXP = /\A(owner=[\w\-]+)|(repo=[\w\-]+)|(file=.*)|(ref=.*)\Z/
 
   def initialize(_tag_name, input, _parse_context)
     super
-    @data = get_data(input.strip)
+
+    stripped_input  = strip_tags(input)
+    unescaped_input = CGI.unescape_html(stripped_input)
+    @params = parsed_input(unescaped_input)
   end
 
   def render(_context)
     ApplicationController.render(
       partial: PARTIAL,
       locals: {
-        owner: @data[:owner],
-        repo: @data[:repo],
-        ref: @data[:ref]
+        params: @params
       },
     )
   end
 
   private
 
-  def get_data(input)
-    items = input.split
-    owner = items.first
-    repo = items.second
-    ref = items.third || "master"
+  def parsed_input(input)
+    if input.split.length > 1
+      params = input.split
+      owner = params.first
+      repo = params.second
+      ref = params.third || "master"
+
+      raise StandardError, I18n.t("liquid_tags.stackery_tag.missing_argument") unless owner && repo
 
-    validate_items(owner, repo)
+      "owner=#{owner}&repo=#{repo}&ref=#{ref}"
+    else
+      match = pattern_match_for(input, [REGISTRY_REGEXP])
+      raise StandardError, I18n.t("liquid_tags.stackery_tag.missing_argument") unless match
 
-    {
-      owner: owner,
-      repo: repo,
-      ref: ref
-    }
+      extract_params(match[:params])
+    end
   end
 
-  def validate_items(owner, repo)
-    return unless owner.blank? || repo.blank?
+  def extract_params(params)
+    params = params.delete("?").split("&")
+    params.filter_map { |param| param if valid_param(param) }.join("&")
```

Comment:
Does this do the same thing?

```suggestion
    params.select { |param| valid_param(param) }.join("&")
```

I think the `filter_map` version may end up creating a bunch of `&&&` when it filters out params. I don't think it'd be _wrong_, necessarily, but that depends on the implementation of Stackery's URL parser when the request is made.